In [1]:
import pandas as pd

In [3]:
dataframe = pd.read_csv("netflix1.csv")

dataframe.head()

,show_id,type,title,director,country,date_added,release_year,rating,duration,listed_in
0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,United States,9/25/2021,2020,PG-13,90 min,Documentaries
1,s3,TV Show,Ganglands,Julien Leclercq,France,9/24/2021,2021,TV-MA,1 Season,"Crime TV Shows, International TV Shows, TV Act..."
2,s6,TV Show,Midnight Mass,Mike Flanagan,United States,9/24/2021,2021,TV-MA,1 Season,"TV Dramas, TV Horror, TV Mysteries"
3,s14,Movie,Confessions of an Invisible Girl,Bruno Garotti,Brazil,9/22/2021,2021,TV-PG,91 min,"Children & Family Movies, Comedies"
4,s8,Movie,Sankofa,Haile Gerima,United States,9/24/2021,1993,TV-MA,125 min,"Dramas, Independent Movies, International Movies"


In [6]:
#q1. How many rows and columns are there in the dataset? 

no_of_rows = dataframe.shape
no_of_rows

(8790, 10)

There are 8790 rows in this dataset.

In [10]:
#q2. How many unique directors are there in the dataset?
unique_dir = len(dataframe['director'].unique())

unique_dir

4528

There are 4528 unique directors in the dataset. 

In [23]:
#q3. Which country produced the maximum number of shows? 
max_shows_country = dataframe['country'].value_counts().idxmax()
max_shows_country

'United States'

United States has made the most shows.

In [18]:
#q4. What is the distribution of shows based on their release year? 
year_dist = dataframe['release_year'].value_counts()
year_dist

2018    1146
2017    1030
2019    1030
2020     953
2016     901
        ... 
1966       1
1959       1
1925       1
1947       1
1961       1
Name: release_year, Length: 74, dtype: int64

The most prolific years have been 2018, 2017, 2019, 2020 and 2016 (in this order).

In [24]:
#q.5 What are the top 5 most common countries in the dataset? 
common_countries = dataframe['country'].value_counts().head()
common_countries

United States     3240
India             1057
United Kingdom     638
Pakistan           421
Not Given          287
Name: country, dtype: int64

It is worth noting that 287 rows don't have a country associated with it.

In [26]:
#q.6 Which genres/categories have the highest frequency?

genres = dataframe['listed_in'].str.split(',').explode()
genre_counts = genres.str.strip().value_counts()

genre_counts.head()


International Movies      2752
Dramas                    2426
Comedies                  1674
International TV Shows    1349
Documentaries              869
Name: listed_in, dtype: int64

Drama is the most important genre.

In [34]:
#q.7 Find the earliest and latest release year for both TV shows as well as movies.

grouped_dataframe = dataframe.groupby('type') 
grouped_year = grouped_dataframe['release_year']

earliest_years = grouped_year.min()
latest_years = grouped_year.max()

print(latest_years)
print(earliest_years)

type
Movie      2021
TV Show    2021
Name: release_year, dtype: int64
type
Movie      1942
TV Show    1925
Name: release_year, dtype: int64


The earliest TV show was released in 1925 and the latest TV show was releases in 2021. <br> The earliest movie was released in 1942 and the latest movie was releases in 2021. 


In [37]:
#q8. Calculate the percentage of shows or movies added to Netflix before 2010.

dataframe['release_year'] = pd.to_numeric(dataframe['release_year']) #converting datatype
shows_before_2010 = (dataframe['release_year'] < 2010).sum() #aggregating number of shows released before 2010
total_shows = len(dataframe) #getting total number of rows
percentage_before_2010 = (shows_before_2010 / total_shows) * 100
percentage_before_2010


15.15358361774744

15.15% shows were released before 2010.

In [41]:
#q9. Calculate the percentage of TV Shows released in the subcontinent.

#first we will extract the shows and then further shortlist shows that were released in Pak, India and Bangaldesh
#do a quick google research on why we used == instead of = when extracting TV shows (next line)
subcontinent_shows = dataframe[(dataframe['type'] == 'TV Show') & (dataframe['country'].str.contains('India|Pakistan|Bangladesh'))]


subcontinent_shows_count = len(subcontinent_shows)
total_tv_shows = (dataframe['type'] == 'TV Show').sum()
percent_sub_shows = (subcontinent_shows_count / total_tv_shows) * 100

percent_sub_shows


16.17867867867868

16.18% shows originated from Pakistan, India or Bangladesh (subcontinent).

In [43]:
#q.10 Order the months from most popular to least popular based on how many TV shows or movies were released in that month. 
dataframe['date_added'] = pd.to_datetime(dataframe['date_added']) #changing datatype to date/time


dataframe['added_month'] = dataframe['date_added'].dt.month_name() #extracting the name of the month


month_counts = dataframe['added_month'].value_counts()


month_count


July         827
December     812
September    769
April        763
October      760
August       754
March        741
January      737
June         728
November     705
May          632
February     562
Name: added_month, dtype: int64

July is the most popular month and February is the least popular month.

In [44]:
#q.11 Bonus question: Based on the analysis above, if someone was to produce a TV show or movie, what would you recommend them?

#we will discuss the answer in the next session.